packages/runner-shared/src/event-manager.js

```diff
@@ -313,6 +321,12 @@ export const eventManager = {
       this._clearAllCookies()
       this._setUnload()
     })
+
+    // The window.top should not change between test reloads, and we only need to bind the message event once
+    // Forward all message events to the current instance of the multi-origin communicator
+    window.top?.addEventListener('message', ({ data, source }) => {
+      Cypress?.primaryOriginCommunicator.onMessage({ data, source })
```

Comment:
ditto on Ryan's comment earlier in this file: https://github.com/cypress-io/cypress/pull/18075/files#r852447833

Reply:
Yeah, I don't think there's any reason `Cypress` would be not be instantiated by the time we get a message via postMessage. I'm going to remove the optional chaining.

Reply:
Addressed in https://github.com/cypress-io/cypress/pull/21137/commits/2ae28a0d55cfb7393d4697dd7bd546e03d6a1bee